### PR TITLE
LPS-170984 Make possible filter ObjectEntries by creatorId

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-170984

### CODE REVIEW

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
@p-albuquerque 

We should cover this case by using integration tests.

[see](https://github.com/liferay/liferay-portal/blob/9a050a6170929f86dd32a8458cb6b850f2308b66/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L1189-L1204)

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -1184,6 +1184,18 @@ public void testGetObjectEntries() throws Exception {
...
 
+		_testGetObjectEntries(
+			HashMapBuilder.put(
+				"filter",
+				() -> {
+					User user = TestPropsValues.getUser();
+
+					return _buildEqualsExpressionFilterString(
+						"creatorId", user.getUserId());
+				}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1195]: @p-albuquerque 

replace to 

````java
_buildEqualsExpressionFilterString("creatorId", _adminUser.getUserId())
````

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -1984,9 +1996,13 @@ private void _assignOrganizationRole(
...
 	private String _buildEqualsExpressionFilterString(
-		String fieldName, String value) {
+		String fieldName, Object value) {
+
+		if (value instanceof String) {
+			return StringBundler.concat("( ", fieldName, " eq '", value, "')");
+		}
 
-		return StringBundler.concat("( ", fieldName, " eq '", value, "')");
+		return StringBundler.concat("( ", fieldName, " eq ", value, ")");
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2005]: @p-albuquerque 

````java
if (value instanceof String) {
  value = "'" + value + "'"
}
````

I think we have a method to surround a string with quotes.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
Directly sent in: https://github.com/brianchandotcom/liferay-portal/pull/132660

**Pull request closed without merging**